explorer/explorer.go

```diff
@@ -574,6 +531,55 @@ func (exp *explorerUI) Store(blockData *blockdata.BlockData, msgBlock *wire.MsgB
 
 	log.Debugf("Got new block %d for the explorer.", newBlockData.Height)
 
+	// Do not run remaining updates when blockchain sync is running.
+	if exp.AreDBsSyncing() {
+		return nil
+	}
+
+	// Simulate the annual staking rate.
+	go func(height int64, sdiff float64, supply int64) {
+		ASR, _ := exp.simulateASR(1000, false, stakePerc,
+			dcrutil.Amount(supply).ToCoin(),
+			float64(height), sdiff)
+		p.Lock()
+		p.HomeInfo.ASR = ASR
+		p.Unlock()
+	}(newBlockData.Height, blockData.CurrentStakeDiff.CurrentStakeDifficulty,
+		blockData.ExtraInfo.CoinSupply) // eval args now instead of in closure
+
+	// Project fund balance, not useful while syncing.
+	if exp.devPrefetch {
+		go exp.updateDevFundBalance()
+	}
+
+	// Trigger a vote info refresh.
+	go exp.voteTracker.Refresh()
+
+	// Politeia updates happen hourly. Thus, if blocks take 5 minutes on average
+	// to mine, then 12 blocks take approximately 1hr.
+	// https://docs.decred.org/advanced/navigating-politeia-data/#voting-and-comment-data
+	if newBlockData.Height%12 == 0 && exp.proposalsSource != nil {
+		// Update the proposal DB. This is run asynchronously since it involves
+		// a query to Politeia (a remote system) and we do not want to block
+		// execution.
+		go func() {
+			err := exp.proposalsSource.CheckProposalsUpdates()
+			if err != nil {
+				log.Errorf("(PoliteiaBackend).CheckProposalsUpdates: %v", err)
+			}
+		}()
+	}
+
+	// Update consensus agendas data every 5 blocks.
+	if newBlockData.Height%5 == 0 {
+		go func() {
+			err := exp.agendasSource.UpdateAgendas()
+			if err != nil {
+				log.Errorf("(agendaBackend).UpdateAgendas: %v", err)
+			}
+		}()
+	}
```

Comment:
Ideally all of these goroutines would a context so they can be cancelled if the next block comes in before they are completed. (TODO)